.github/workflows/pull_request_validation.yml

```diff
@@ -27,3 +27,5 @@ jobs:
         run: ruff format --check
       - name: Lint Check
         run: ruff check
+      - name: Type Check
+        run: pyright
```

Comment:
is this a utility for static checks?
is this needed for sqlite or just happens to be in this patch?

Reply:
yeah this is static analysis. used it to help with the data refactor